src/ext/issuance.js

```diff
@@ -312,23 +312,18 @@ function verifyProofAndStoreTokens(url, tabId, tokens, issueResp, commitments) {
  * @return {Object} Object containing commitment data
  */
 function retrieveCommitments(xhr, version) {
-    let commG;
-    let commH;
-    const respBody = xhr.responseText;
-    const resp = JSON.parse(respBody);
-    const comms = resp[commitmentsKey()];
+    const resp = JSON.parse(xhr.responseText);
+    const comms = resp[getConfigName()];
+
     version = checkVersion(version);
-    if (comms) {
-        if (dev()) {
-            commG = comms["dev"]["G"];
-            commH = comms["dev"]["H"];
-        } else {
-            commG = comms[version]["G"];
-            commH = comms[version]["H"];
-        }
+    const cmt = comms[version];
+    if (typeof cmt === "undefined") {
+        throw new Error("[privacy-pass]: Retrieved version: " + version + " not available.");
     }
-
-    return {G: commG, H: commH};
+    if (cmt.sig) {
```

Comment:
I'm a little worried about only verifying if there is a signature field but I'm not sure what the best alternative would be.

Perhaps we could an extra member to each of the configs in config.js that specifies whether signatures should be verified by default, we could set the CF config to default to true, and the HC config to default to false until they provide signatures?

I'm not sure what the best approach to still allowing version `"1.0"` to work for CF would be though. I guess we could provide an extra member in the configs that specifies trusted versions (that would include "1.0" for CF?). This would mean that we would structure the commitments members as:
```
cfConfig["commitments"] = {
  "verification-key": <public-key>,
  "verify": true,
  "trusted": ["1.0"],
}
```
and for hCaptcha:
```
hcConfig["commitments"] = {
  "verification-key": null,
  "verify": false,
}
```
Or we could just provide a signature for version 1.0 in a future change, but that would mean the master branch would not work for a while. Do you have an opinion?

Reply:
I think the configuration is getting a lot of flags, which adds more complexity to the program.
 I would prefer something simpler. I think that the presence of signatures in commitments must be enforced. 
This can be controlled by the version of our implementations.
example: version 1.xx means signature is optional.
 version 2.xx means signature is required.


Assume that v2.0 is a big step from v1.0 then it is ok to work on an alternate branch (for not breaking master). and the release day make the merge with master.

Reply:
You are right. Let's agree on a path to follow.